src/librustc_middle/ty/context.rs

```diff
@@ -1120,7 +1120,18 @@ impl<'tcx> TyCtxt<'tcx> {
             consts: common_consts,
             extern_crate_map: resolutions.extern_crate_map,
             trait_map,
-            export_map: resolutions.export_map,
+            export_map: resolutions
```

Comment:
`export_map` in `GlobalCtxt` can use `LocalDefId` instead of `HirId` as well.

Reply:
So i tried that but it implies changing the query `module_exports` to return `LocalDefId`s which I think is fine, but further down it also changes encoding of the module metadata in `EncodeContext::encode_info_for_mod` which i don't know if its fine or not. I have the feeling it will impact incremental compilation?

Reply:
Changing `export_map` only requires changing `module_exports` to take `LocalDefId` (which is certainly ok), not return `LocalDefId`s (which should also be ok, but I'm not sure about the complexity of the involved changes).

Reply:
Ah, ok, there's a mismatch between both keys and values of the map.

Reply:
Could you change `type ExportMap` to use `LocalDefId` instead of `DefId` first of all?
Then use it in `GlobalCtxt`.

Reply:
Yeah, `module_exports` can both take and return `LocalDefId`s.

Reply:
Oh you were talking about only changing map key from `DefId` to `LocalDefId`? I thought you meant changing the value type. What should I change then? I am a bit lost. I can change both I guess, but changing the value type means allocating a `Vec<Export<HirId>>` during encoding of the metadata, which might or might not be fine.

Reply:
>changing the value type means allocating a `Vec<Export<HirId>>` during encoding of the metadata

Why?

Reply:
> changing the value type means allocating a Vec<Export<HirId>> during encoding of the metadata

No, in fact this part is good, I misunderstood the errors. Returning `LocalDefId` makes more sense and is straightforward.

Taking `LocalDefId` on the contrary might not be the right thing to do because it forces calls to `expect_local` in `librustc_resolve` because `ModuleKind` is storing `DefId` not `LocalDefId`.